sda-commons-server-key-mgmt/src/main/java/org/sdase/commons/keymgmt/KeyMgmtBundle.java

```diff
@@ -0,0 +1,181 @@
+package org.sdase.commons.keymgmt;
+
+import io.dropwizard.Configuration;
+import io.dropwizard.ConfiguredBundle;
+import io.dropwizard.setup.Bootstrap;
+import io.dropwizard.setup.Environment;
+import java.io.IOException;
+import java.util.*;
+import org.sdase.commons.keymgmt.config.KeyMgmtConfig;
+import org.sdase.commons.keymgmt.config.KeyMgmtConfigProvider;
+import org.sdase.commons.keymgmt.model.KeyDefinition;
+import org.sdase.commons.keymgmt.model.KeyMappingModel;
+import org.sdase.commons.keymgmt.model.ValueMappingModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KeyMgmtBundle<T extends Configuration> implements ConfiguredBundle<T> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KeyMgmtBundle.class);
+  private final KeyMgmtConfigProvider<T> configProvider;
+
+  private Map<String, KeyDefinition> keys;
+  private Map<String, KeyMappingModel> mappings;
+
+  public static InitialBuilder builder() {
+    return new Builder<>();
+  }
+
+  private KeyMgmtBundle(KeyMgmtConfigProvider<T> configProvider) {
+    this.configProvider = configProvider;
+  }
+
+  @Override
+  public void initialize(Bootstrap<?> bootstrap) {
+    // no init
+  }
+
+  @Override
+  public void run(T configuration, Environment environment) throws IOException {
+    KeyMgmtConfig config = configProvider.apply(configuration);
+
+    keys = ModelReader.parseApiKeys(config.getApiKeysDefinitionPath());
+    mappings = ModelReader.parseMappingFile(config.getMappingDefinitionPath());
+  }
+
+  /**
+   * Maps a api specific <code>value</code> for a <code>key</code> to the implementation specific
+   * value
+   *
+   * @param key the key identifier
+   * @param value the value to be mapped
+   * @return Optional containing value or empty if no mapping exists
+   */
+  public Optional<String> mapFromApiToImpl(String key, String value) {
```

Comment:
As the bundle itself is the mapping service, can you introduce an interface like `ConfigurableKeyMapper` implemented by this `KeyMgmtBundle` so that consuming services are not distracted with access to the methods from `ConfiguredBundle`?

Reply:
ok for me

Reply:
provide now just a `createMapper` and `createKeyManager` method in the bundle